### PR TITLE
Uses powershell for windows

### PR DESCRIPTION
**Pull request by scriptnull**
https://github.com/Shippable/reqProc/issues/224

This results in the following output in my current dev setup
image

### CODE REVIEW

**Comment by rageshkrishna** [comment]
I think this needs to be done in a more generic way. There should be one point in the code that decides whether script extensions are going to be `.ps1` or `.sh` (or something else). Everything that needs to look up or generate a script should use this global setting.

File: job/handlers/executeScript.js [hunk 1]
```diff
@@ -57,7 +57,13 @@ function _executeTask(bag, next) {
   var who = bag.who + '|' + _executeTask.name;
   logger.debug(who, 'Inside');
 
-  var exec = spawn('/bin/bash', ['-c', bag.scriptPath + ' 2>&1'], bag.options);
+  var exec;
+  if (global.config.shippableNodeOperatingSystem === 'WindowsServer_2016') {
+    exec = spawn('powershell', [bag.scriptPath], bag.options);
```
Comment (shrivara) [inline job/handlers/executeScript.js]: We redirect the stderr stream to stdout below as we want to handler everything in the stdout event. Is this not required here?

File: job/scriptsGen/normalizeSteps.js [hunk 2]
```diff
@@ -170,7 +170,8 @@ function __generateRuntimeInfo(task, buildJobId, buildScriptsDir,
     taskName: task.name || util.format('task_%s', task.taskIndex),
     isTaskInContainer: task.runtime.container
   });
-  task.taskScriptFileName = util.format('%s_task_%s.sh', group, task.taskIndex);
+  task.taskScriptFileName = util.format('%s_task_%s.%s', group,
+      task.taskIndex, global.config.scriptExtension);
```
Comment (shrivara) [inline job/scriptsGen/normalizeSteps.js]: Two spaces.

File: _common/micro/setupMS.js [hunk 3]
```diff
@@ -45,6 +45,12 @@ function setupMS(params) {
     process.env.SHIPPABLE_NODE_OPERATING_SYSTEM
   );
 
+  if (global.config.shippableNodeOperatingSystem === 'WindowsServer_2016') {
```
Comment (shrivara) [inline _common/micro/setupMS.js]: No braces.

File: job/handlers/executeScript.js [hunk 4]
```diff
@@ -57,7 +57,13 @@ function _executeTask(bag, next) {
   var who = bag.who + '|' + _executeTask.name;
   logger.debug(who, 'Inside');
 
-  var exec = spawn('/bin/bash', ['-c', bag.scriptPath + ' 2>&1'], bag.options);
+  var exec;
+  if (global.config.shippableNodeOperatingSystem === 'WindowsServer_2016') {
```
Comment (shrivara) [inline job/handlers/executeScript.js]: No braces.

**Pull request merged**